Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -902,19 +945,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
     updateStates();
 }
 
+void MediaPlayerPrivateGStreamerMSE::unmarkEndOfStream()
+{
+    GST_DEBUG("Unmarking end of stream");
+    m_eosPending = false;
+}
+
 MediaTime MediaPlayerPrivateGStreamerMSE::currentMediaTime() const
 {
+    MediaTime cachedPosition = MediaTime::createWithFloat(m_cachedPosition);
```

Comment:
no float here!

Reply:
That would require refactoring the m_cachedPosition usage (my media time patch never made it into these branches, or upstream). I'm not going to refactor this while merging.

Reply:
Ok, but please, do not lose track of that cause I think it is an important change.